packages/commonjs/src/index.js

```diff
@@ -34,7 +34,7 @@ import {
 } from './transform';
 
 export default function commonjs(options = {}) {
-  const extensions = options.extensions || ['.js'];
+  const extensions = options.extensions || ['.js', '.cjs'];
```

Comment:
q: what exact scenario has been failing and why? from what I understand `.cjs` (nor `.mjs`) can ever be implicit (please correct me if I'm wrong), so what `@rollup/plugin-commonjs` has been doing that it didn't handle this correctly? 🤔

Reply:
Looking at `getCandidates` here:
https://github.com/rollup/plugins/blob/8c49103deb620b26d0d019ee19c3d1420a8933dc/packages/commonjs/src/resolve-id.js#L23
it seems that explicit `.cjs` should always work as it's most likely the `resolved` variable here unless I miss something.

Reply:
I included .mjs erroneously, but this PR was prompted by the behavior in #523, where the output would have a `module.exports` declaration at the top as well as ES2015 `export`s at the bottom when rolling in a .cjs file.

Reply:
Ah, I think the problem is that those extensions have a double meaning:
1. They allow for implicit resolution
2. They are being used as a filter: https://github.com/rollup/plugins/blob/5fa15590e452d568239a6d8e1c7018865c027fe4/packages/commonjs/src/index.js#L192

The latter is what the problem is here.

Reply:
But you are right, `.cjs` should not be allowed for implicit resolution by default as this contradicts the algorithm in Node.

Reply:
@christiantjl please review these recent comments

Reply:
@shellscape I'm pretty new to Rollup and Node in general, could you dumb down these comments for me? Is the issue that my change will extend to like, `require('./dir/root')` catching `./dir/root.cjs` when that would conflict with Node logic? Any advice on remedying?

Reply:
> Is the issue that my change will extend to like, require('./dir/root') catching ./dir/root.cjs when that would conflict with Node logic?

Yes, this should not resolve by default to `.cjs`. 

> Any advice on remedying?

The applied change is not correct - as @lukastaegert has observed it seems that `extensions` are used now by 2 separate places in the logic. You have added `.cjs` to the first one incorrectly (allowing for implicit resolution) but actually `.cjs` should just be passed through this condition at all times:
https://github.com/rollup/plugins/blob/5fa15590e452d568239a6d8e1c7018865c027fe4/packages/commonjs/src/index.js#L191-L196
As we know that `.cjs` has to be in the commonjs format.

Reply:
OK, thank you very much for explaining @Andarist. I will update shortly.